CMake Release build fails on Ubuntu 18.04

**Debug** and **RelWithDebInfo** builds are working

## How to replicate:
1. cd into git repo on master
2. `mkdir bld && cd bld && cmake .. -DCMAKE_BUILD_TYPE=Release && make -j4`

## This is the warning->error:
```
[ 41%] Building CXX object lib/gamelib/CMakeFiles/gamelib.dir/audp_lexer.cpp.o
Scanning dependencies of target netplay
[ 41%] Building CXX object lib/netplay/CMakeFiles/netplay.dir/netjoin_stub.cpp.o
Scanning dependencies of target ivis-opengl
[ 42%] Building CXX object lib/ivis_opengl/CMakeFiles/ivis-opengl.dir/bitimage.cpp.o
lex.audp_.c: In function ‘void audp_restart(FILE*)’:
lex.audp_.c:1553:20: error: potential null pointer dereference [-Werror=null-dereference]
lex.audp_.c:1552:19: error: potential null pointer dereference [-Werror=null-dereference]
cc1plus: all warnings being treated as errors
make[2]: *** [lib/gamelib/CMakeFiles/gamelib.dir/audp_lexer.cpp.o] Error 1
lib/gamelib/CMakeFiles/gamelib.dir/build.make:82: recipe for target 'lib/gamelib/CMakeFiles/gamelib.dir/audp_lexer.cpp.o' failed
make[1]: *** [lib/gamelib/CMakeFiles/gamelib.dir/all] Error 2
```

### cmake config log:
https://pastebin.com/a4FaV9mU
### make log (with the error):
https://pastebin.com/uTt2FK6P

@bjorn-ali-goransson relplicated this behavior on their machine

Its not a big issue ATM since the travis build is compiling in RelWithDebInfo since this commit (changed from Debug) https://github.com/Warzone2100/warzone2100/commit/45615471384479454dfa133ccf6eed6a344abc10

Ive shortly checked out the code in audp_lexer.cpp and it seems it has been generated by a scanner generator (flex). It has not been touched in a while. 

Possible solutions could be (if working Release builds on Ubuntu are a goal):
- re-generate the scanner with different flex flags to make it more robust on release builds. i have found this forum post about the generators in the git history http://forums.wz2100.net/viewtopic.php?p=88749
- not treat this warning as an error `lex.audp_.c:1553:20: error: potential null pointer dereference [-Werror=null-dereference]` (change would probably need to be made here https://github.com/Warzone2100/warzone2100/blob/0861c922d8891be1f1b878da617c8544810cd80d/CMakeLists.txt#L497 )

Default Ubuntu 18.04 compiler flags for the builds are as follows:
```
  string(APPEND CMAKE_${lang}_FLAGS_INIT " ")                                                        //default
  string(APPEND CMAKE_${lang}_FLAGS_DEBUG_INIT " -g")                                      
  string(APPEND CMAKE_${lang}_FLAGS_MINSIZEREL_INIT " -Os -DNDEBUG")
  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O3 -DNDEBUG")
  string(APPEND CMAKE_${lang}_FLAGS_RELWITHDEBINFO_INIT " -O2 -g -DNDEBUG")
```

--- a/lib/gamelib/audp_lexer.cpp
+++ b/lib/gamelib/audp_lexer.cpp
@@ -1,6 +1,28 @@
+/*
+	This file is part of Warzone 2100.
+	Copyright (C) 2005-2020  Warzone 2100 Project
+
+	Warzone 2100 is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+
+	Warzone 2100 is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with Warzone 2100; if not, write to the Free Software
+	Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+*/
 
 #if defined(__clang__)
 	#pragma clang diagnostic ignored "-Wunneeded-internal-declaration" // warning: function 'yyinput' is not needed and will not be emitted
+#elif defined(__GNUC__)
+	#if 7 <= __GNUC__
+		#pragma GCC diagnostic ignored "-Wnull-dereference"
+	#endif
 #endif
 
 #line 3 "lex.audp_.c"

--- a/lib/ivis_opengl/screen.cpp
+++ b/lib/ivis_opengl/screen.cpp
@@ -526,6 +526,18 @@ void screen_SetRandomBackdrop(const char *dirname, const char *basename)
 	}
 	PHYSFS_freeList(rc);
 
+	if (names.empty())
+	{
+		std::string searchPath = dirname;
+		if (!searchPath.empty() && searchPath.back() != '/')
+		{
+			searchPath += "/";
+		}
+		debug(LOG_FATAL, "Missing files: \"%s%s*\" - data files / folders may be corrupt.", searchPath.c_str(), basename);
+		abort();
+		return;
+	}
+
 	// pick a random name from our vector of names
 	int ran = rand() % names.size();
 	std::string full_path = std::string(dirname) + names[ran];

